incompatibility/incompatibility-report/incomp-scanning.go

```diff
@@ -39,10 +41,32 @@ func CreateIncompReport(env *plugins.Environment, reportType Report) {
 
 // Scan for incompatibilities in an OpenAPI document
 func ScanIncompatibilities(document *openapiv3.Document) *IncompatibilityReport {
-	var incompatibilities []*Incompatibility
 
-	if document.Servers != nil {
-		incompatibilities = append(incompatibilities, &Incompatibility{TokenPath: []string{"servers"}, Classification: "SERVERS"})
+	paths, err := knownIncompatibilityPaths()
+	if err == 1 {
+		return &IncompatibilityReport{}
+	}
+	IncompReport, err := paths.compile(document)
+	if err == 1 {
+		return &IncompatibilityReport{}
 	}
-	return &IncompatibilityReport{Incompatibilities: incompatibilities}
+	return IncompReport
+}
+
+// Function to get path representations of
+func knownIncompatibilityPaths() (PathOperation, int) {
```

Comment:
Returning an `int` as the error is not go practice. Use the normal Golang error. Apply this to all the error returns.

Reply:
now makes use of golang error